envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -146,8 +146,8 @@ class SnapshotUpdater(
                 updateSnapshotForGroup(group, result.xdsSnapshot)
             } else {
                 meterRegistry.counter("snapshot-updater.communication-mode.errors").increment()
-                logger.error("Requested snapshot for ${group.communicationMode.name} mode," +
-                    " but it is not here. This should never happen")
+                logger.error("Handling Envoy with not supported communication mode should be rejected before." +
```

Comment:
If I understand correctly @chemicL comment, we should keep the part: `Requested snapshot for ${group.communicationMode.name} mode," +
                    " but it is not here`
and only change "This should never happen" bit for the new one. 

So I think in the end it should be:
> Requested snapshot for ${group.communicationMode.name} mode, but it is not here. Handling Envoy with not supported communication mode should be rejected before. Please report this to EC developers.

IMO knowing which snapshot is not found is rather important.

LMK @chemicL

Reply:
Right, I missed that part also changed. I agree with Marcin, we need the communication mode in the log too.
BTW `should be rejected before` -> `should have been rejected before`.